.github/workflows/heroku-deploy.yml

```diff
@@ -0,0 +1,34 @@
+name: Heroku Deploy
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+# `types: [closed]` above means that this step runs even when a PR is
+# closed without merging.
+# This `if` statement ensures we only run this step when the PR is merged.
+if: ${{ github.event.pull_request.merged == true }}
```

Comment:
So this all together means that we only deploy when we close (merge?) a PR into main?

Reply:
Yes, you're exactly right.

> close (merge?)

Yeah, there's no `merge` event, so we have to check for when PRs are closed (which means closed or merged) then confirm that it's a merge.